src/org/usfirst/frc/team2399/robot/subsystems/DriveTrain.java

```diff
@@ -4,35 +4,218 @@
 		leftBackTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
+		leftMiddleTalon.set(leftSpeed * RobotMap.DRIVETRAIN_FORWARD_LEFT_CONSTANT);
 	}
+	
 
-	public void driveRight(double rightSpeed) {
+	public void driveRightVelocity(double rightSpeed) {
+		rightFrontTalon.changeControlMode(TalonControlMode.Speed);
+		
+		rightFrontTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightBackTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightMiddleTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+	}
+	
+	public void driveRightPercent(double rightSpeed) {
+		rightFrontTalon.changeControlMode(TalonControlMode.PercentVbus);
+		
 		rightFrontTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
 		rightBackTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
+		rightMiddleTalon.set(rightSpeed * RobotMap.DRIVETRAIN_FORWARD_RIGHT_CONSTANT);
 	}
 
+	
+	
+	/**
+	 * Gets the current position of the robot
+	 * Multiplied by the circumference of the wheel for scaling
+	 * @return
+	 */
+	public double getLeftPosition(){
+		return leftFrontTalon.getPosition() * RobotMap.DRIVETRAIN_WHEEL_CIRCUMFERENCE_CONSTANT;
+	}
+	
+	public double getRightPosition(){
+		return rightFrontTalon.getPosition() * RobotMap.DRIVETRAIN_WHEEL_CIRCUMFERENCE_CONSTANT;
+	}
+	
+	/**
+	 * Methods that get/set the left and right positions of where we want to be
+	 * setPosition zeros out the encoder data, so that we start at 0
+	 */
+	
+	public void setLeftDesiredPosition(double goalDistance){
+		leftFrontTalon.setPosition(0);
+		this.goalDistance = goalDistance;
+	}
+	
+	public void setRightDesiredPosition(double goalDistance){
+		rightFrontTalon.setPosition(0);
+		this.goalDistance = goalDistance;
+	}
+	
+	public double getLeftDesiredDistance()
+	{
+		return goalDistance;
+	}
+	
+	public double getRightDesiredDistance()
+	{
+		return goalDistance;
+	}
+	
+
+	
+	/**
+	 * Methods that calculate the error in distance from where we are to where we want to be
+	 */
+	
+	public double calculateLeftDistanceError(){
+		return rightFrontTalon.getError();
+		
+	}
+	
+	public double calculateRightDistanceError(){
+		return leftFrontTalon.getError();
+		
+	}
+	
+	/**
+	 * Methods to increment, decrement, and get the distance constant
+	 * Method takes in the talon the P constant is being set for, gets the current P, and sets
+	 * the P to the current P + the increment/decrement constant
+	 */
+	
+	public void incrementDistanceConstant(CANTalon currentTalon){
+		double currentPConstant = currentTalon.getP();
+		currentTalon.setP(currentPConstant += RobotMap.DISTANCE_INCREMENT_CONSTANT);
+	}
+	
+	public void decrementDistanceConstant(CANTalon currentTalon){
+		double currentPConstant = currentTalon.getP();
+		currentTalon.setP(currentPConstant-= RobotMap.DISTANCE_DECREMENENT_CONSTANT);
+	}
+	
+	public double getDistanceConstant(CANTalon currentTalon){
```

Comment:
Is this the right name for a method that returns the P-constant? (Genuine question, as I am not sure)